### PR TITLE
Update release process for bktec

### PR DESCRIPTION
**Pull request by amybiyuliu**
### Description

Update Goreleaser: 
- The binary name is called `bktec`
- The binary will be published to new dockerhub and package cloud repositories (with `test-engine-client` in the url) 
- The binary will be published to homebrew as `bktec`

### Testing

Verified the change locally by runing `goreleaser release --clean --skip publish --snapshot`. The binaries are created with name `bktec`

After merging this PR we can release v1.0.0-rc.1 to verify the actual release works

### CODE REVIEW

File: .goreleaser.yaml
Comment (nprizal) [inline .goreleaser.yaml:25]:
```suggestion
    description: "Buildkite Test Engine Client"
```

File: .goreleaser.yaml
Comment (nprizal) [inline .goreleaser.yaml:91]:
```suggestion
    description: Buildkite Test Engine Client
```

**Review by nprizal — APPROVED** [review]
Paired with @amybiyuliu and this looks good to me. @niceking Just one thing to confirm, we want the binary name to `bktec`, is that right? So the usage will be
```
export BUILDKITE_TEST_ENGINE_API_ACCESS_TOKEN=token
export BUILDKITE_TEST_ENGINE_SUITE_SLUG=my-suite
# export other env vars here

bktec
```

File: .buildkite/steps/upload-linux-packages.sh
Comment (niceking) [inline .buildkite/steps/upload-linux-packages.sh:5]: For thing that aren't already namespaced under "buildkite", I reckon we should call it "buildkite-test-engine-client"

File: .goreleaser.yaml
Comment (niceking) [inline .goreleaser.yaml:4]: As above

File: .buildkite/steps/upload-linux-packages.sh
Comment (nprizal) [reply .buildkite/steps/upload-linux-packages.sh:5]: Package cloud registry is stored within `buildkite` namespace. See this [example](https://buildkite.com/organizations/buildkite/packages/registries/test-splitter-deb/packages/019106fc-6d20-781b-816e-74c39e3dee99) from the old registry.

File: .goreleaser.yaml
Comment (amybiyuliu) [inline .goreleaser.yaml:2]: Add `project_name` here as I found the binary being put into `dist/test-splitter_*` on the test engine client pipeline build. When testing locally it puts the binary to `dist/test-engine-client*`


> building binaries
> --
>   | 2024-09-16 16:51:08 ACST | • partial build                                  match=target=darwin_amd64
>   | 2024-09-16 16:51:08 ACST | • building                                       binary=dist/test-splitter_darwin_amd64_v1/bktec

The example build: https://buildkite.com/buildkite/test-engine-client/builds/1003#0191f9b3-4701-4d79-951a-7c8a44704f89


@nprizal is this a sensible change?

**Comment by amybiyuliu** [comment]
I made some minor changes to this PR in the afternoon, would be nice to have another review.
I'm not sure whether this is blocking release for https://github.com/buildkite/test-engine-client/pull/188 
@wooly please feel free to take over this, or merge this PR if it's blocking your release

File: .goreleaser.yaml
Comment (nprizal) [reply .goreleaser.yaml:2]: Ah, interesting! g
oreleaser get the project name from repository name by default, we have renamed the repository, but the pipeline still referring to the old repo name (see image below). I think explicitly setting the project name is a good idea here.
Screenshot 2024-09-17 at 9 14 26 AM

**Review by nprizal — APPROVED** [review]

File: .goreleaser.yaml
Comment (nprizal) [inline .goreleaser.yaml:5]:
```suggestion
  name_template: Test Engine Client v{{.Version}}
```
I don't think this is necessary. This is the title for GitHub release. The repo is already under Buildkite namespace.

**Pull request merged**
